File: 1._A._Latihan_Random_Coding/Demo_Coding/Demo_Error_Handling/Fungsi_Demo_Error_Handling.js
// Fungsi Error Handling

console.log ("\n Fungsi Error Handling \n")


function cekAngka (a) {
    try {
        if (a < 0) {
            throw ("Minus")
            console.log (`Angka salah ${a}`)
        }
        
        else {
            console.log (`Angka benar ${a}`)
        }
        
    }
    
    catch (Error) {
        console.log (`Gak boleh minus, a = ${a}`)
    }
}

cekAngka (-5)


console.log ("\n --- Batas --- \n")





function dasar (b) {
    try {
        if (b < 0) {
            throw ("Minus")
            console.log (`Angka Salah ${b}`)
        }
        
        else {
            console.log (`Angka benar, angka = ${b}`)
        }
    }
    
    catch (Error) {
        console.log (`Gak boleh minus, angka = ${b}`)
    }
}

dasar (10)

File: 1._A._Latihan_Random_Coding/Demo_Coding/Demo_Error_Handling/Fungsi_Demo_Error_Handling.py
# Fungsi Error Handling 

print ("\n Fungsi Error Handling \n")


def cek_angka (a):
    try:
        if a < 0:
            raise ("Minus")
            print (f"Angka salah {a}")
            
        else:
            print (f"Angka benar")
            
    except:
        print (f"Gak boleh minus, a = {a}")
    
    
cek_angka (-5)


print ("\n --- Batas --- \n")




print ("\n Fungsi Error Handling 2 \n")

def dasar (b):
    try:
        if b < 0:
            raise ("Minus")
            print (f"Angka salah, angka = {b}")
            
        else:
            print (f"Angka benar, angka = {b}")
            
    except:
        print (f"Gak boleh minus, angka = {b}")
        
        
print ("\n --- Batas --- \n")
